File: Week_1/OOPS/LibraryInterfaceDemo.java
class LibraryInterfaceDemo {
    public static void main(String[] args) {
        // Test case #1:
        KidUser kid = new KidUser();
        kid.registerAccount(10);
        kid.registerAccount(18);
        kid.requestBook("Kids");
        kid.requestBook("Fiction");

        // Test case #2:
        AdultUser adult = new AdultUser();
        adult.registerAccount(5);
        adult.registerAccount(23);
        adult.requestBook("Kids");
        adult.requestBook("Fiction");
    }
}

File: Week_1/OOPS/in_c/LibraryInterfaceDemo.c
#include <stdlib.h>
#include "KidUser.h"
#include "AdultUser.h"

int main() {
    // Test case #1:
    KidUser *kid = createKidUser();
    LibraryUser *kidUser = (LibraryUser *)kid;
    kidUser->registerAccount(kidUser, 10);
    kidUser->registerAccount(kidUser, 18);
    kidUser->requestBook(kidUser, "Kids");
    kidUser->requestBook(kidUser, "Fiction");

    // Test case #2:
    AdultUser *adult = createAdultUser();
    LibraryUser *adultUser = (LibraryUser *)adult;
    adultUser->registerAccount(adultUser, 5);
    adultUser->registerAccount(adultUser, 23);
    adultUser->requestBook(adultUser, "Kids");
    adultUser->requestBook(adultUser, "Fiction");

    free(kid);
    free(adult);
    return 0;
}
